src/NServiceBus.Core/Serializers/MessageDeserializerResolver.cs

```diff
@@ -12,14 +12,17 @@ class MessageDeserializerResolver
 
         public MessageDeserializerResolver(IEnumerable<IMessageSerializer> messageSerializers, Type defaultSerializerType)
         {
-            Guard.AgainstNull("defaultSerializerType", defaultSerializerType);
-
             serializersMap = messageSerializers.ToDictionary(key => key.ContentType, value => value);
             defaultSerializer = serializersMap.Values.Single(serializer => serializer.GetType() == defaultSerializerType);
         }
 
         public IMessageSerializer Resolve(string contentType)
         {
+            if (contentType == null)
```

Comment:
i thought about that but ruled it out because the logic of how to select the serializer should clearly exist inside MessageDeserializerResolver and not inside the caller